src/components/fields/MultiActiveArrayField.tsx

```diff
@@ -0,0 +1,241 @@
+import ReactContext from "../../ReactContext";
+import * as PropTypes from "prop-types";
+import { getUiOptions, ReactUtils } from "../../utils";
+import merge from "deepmerge";
+import { beforeAdd } from "../templates/ArrayFieldTemplate";
+import { copyItemFunction } from "./ArrayField";
+import { AccordionArrayFieldTemplate, getPopupDataPromise } from "./SingleActiveArrayField";
+import { FormContext } from "../LajiForm";
+
+interface State {
+	activeIdx: number[];
+	scrollHeightFixed: number;
+	popups: any;
+}
+
+@BaseComponent
+export default class MultiActiveArrayField extends React.Component<FieldProps<JSONSchemaObject>, State> {
+	static contextType = ReactContext;
+	static propTypes = {
+		uiSchema: PropTypes.shape({
+			"ui:options": PropTypes.shape({
+				renderer: PropTypes.oneOf(["accordion"]),
+				activeIdx: PropTypes.arrayOf(PropTypes.number)
+			})
+		}),
+		schema: PropTypes.shape({
+			type: PropTypes.oneOf(["array"])
+		}).isRequired,
+		formData: PropTypes.array
+	}
+
+	deleteButtonRefs = {};
+	deleteButtonRefSetters = {};
+	localContext?: FormContext;
+	localContextKey?: number;
+	mounted = false;
+
+	constructor(props: FieldProps<JSONSchemaObject>) {
+		super(props);
+		this.deleteButtonRefs = {};
+		this.deleteButtonRefSetters = {};
+		this.state = {
+			activeIdx: this.getInitialActiveIdx(props),
+			scrollHeightFixed: 0,
+			...this.getStateFromProps(props),
+			popups: {}
+		};
+		const id = `${this.props.idSchema.$id}`;
+		this.props.formContext.globals[`${id}.activeIdx`] = this.state.activeIdx;
+	}
+
+	getInitialActiveIdx = (props: FieldProps<JSONSchemaObject>): number[] => {
+		const {formData} = props;
+		const formDataLength = (formData || []).length;
+		return [...Array(formDataLength).keys()];
+	}
+
+	componentDidMount() {
+		this.mounted = true;
+		this.updatePopups(this.props);
+	}
+
+	componentWillUnmount() {
+		this.mounted = false;
+	}
+
+	UNSAFE_componentWillReceiveProps(props: FieldProps<JSONSchemaObject>) {
+		this.setState(this.getStateFromProps(props));
+		this.updatePopups(props);
+	}
+
+	componentDidUpdate() {
+		this.props.formContext.globals[`${this.props.idSchema.$id}.activeIdx`] = this.state.activeIdx;
+	}
+
+	shouldComponentUpdate(prevProps: FieldProps<JSONSchemaObject>, prevState: State) {
+		if ((this.state.scrollHeightFixed && !prevState.scrollHeightFixed)
+			|| this.state.scrollHeightFixed && this.state.scrollHeightFixed !== prevState.scrollHeightFixed
+		) {
+			return false;
+		}
+		return true;
+	}
+
+	updatePopups = (props: FieldProps<JSONSchemaObject>) => {
+		const {popupFields} = getUiOptions(this.props.uiSchema);
+		let {popups} = this.state;
+		let count = 0;
+		if (popupFields) props.formData.forEach((item: any, idx: number) => {
+			getPopupDataPromise(props, item).then(popupData => {
+				count++;
+				popups  = {...popups, [idx]: popupData};
+				if (this.mounted && count === props.formData.length) this.setState({popups});
+			}).catch(() => {
+				count++;
+			});
+		});
+	}
```

Comment:
pretty much all of this can be removed (expect `getInitialActiveIdx`), since the `render()` doesn't render popups and it doesn't use the `Affix` component which affects the `topOffset`. Even in the future the `Affix` will never be usable, since it wouldn't make sense to use it when multiple panels can be open.